t/test01.t

```diff
@@ -83,14 +100,17 @@ sub run_zonemaster_test_with_backend_API {
     }
 
     use_ok( 'Zonemaster::Backend::TestAgent' );
-    Zonemaster::Backend::TestAgent->new( { dbtype => "SQLite", config => $config } )->run( $hash_id );
+    Zonemaster::Backend::TestAgent->new( { dbtype => "$db_backend", config => $config } )->run( $hash_id );
 
     Zonemaster::Backend::TestAgent->reset() unless ( $ENV{ZONEMASTER_RECORD} );
 
+    sleep( 5 ) unless ( $ENV{ZONEMASTER_RECORD} );
```

Comment:
I tried to think about this call to sleep. My guess is that the sleep here is intended to give the TestAgent time to pick up the request. But if that is the case, I can't see why it's only needed if we aren't recording. Am I getting something wrong?

Reply:
Oh, I think I misunderstand what and how this `ZONEMASTER_RECORD` environment variable worked.

Now, from my understanding of the code if this variable is unset or egal to `0`, the test will use the data in `t/test01.data`. Otherwise, it will run all the tests as done in production. Is that so ?

And a subsidiary question, when setting `ZONEMASTER_RECORD=1`, will it actually « record » something ? (or should it « record » something ?

Reply:
Yes, 
```
if ( $ENV{ZONEMASTER_RECORD} ) {
	Zonemaster::Engine->save_cache( $datafile );
}
```

Reply:
Yes, the idea is to make the unit tests run with recorded data by default, so as to make them independent of network connectivity and changes to the internet. When you update Engine so it makes new queries that isn't covered by the recorded data, you need to rerecord it. It's a pretty neat feature!

For completeness I might also add that ZONEMASTER_RECORD also counts as false when it's set to the empty string.

Reply:
Ok. I have a better view now. Thanks.

These `sleep()` calls look like they are needed to wait for the tests to finish. However running the code with and without the `ZONEMASTER_RECORD` set to `1`, it looks like the `TestAgent->run()` is blocking. So there might be no need to call these `sleep()`. Unless the code is updated to run the TestAgent in a subprocess (and therefore may be killed if it hangs).

Reply:
Right. It makes sense that TestAgent is single-threaded. We're using Parallel::ForkManager to manage a set of worker processes, each one of which is using TestAgent.

If we can do without the sleeping now I say good riddance. If we need it later we should probably first look for alternatives before adding it back.

Reply:
The loop has been removed in 741b75b